stingray/pulse/search.py

```diff
@@ -84,19 +98,25 @@ def epoch_folding_search(times, frequencies, nbin=128, segment_size=5000,
         correct for the exposure (Use it if the period is comparable to the
         length of the good time intervals.)
     """
-    if expocorr or not HAS_NUMBA:
+    if expocorr or not HAS_NUMBA or isinstance(weights, collections.Iterable):
```

Comment:
I'm assuming you test here for `weights` being a list or numpy.array? Are there cases where this can go wrong? What else could be an iterable? A dictionary? A pandas.DataFrame? An astropy.fits object? Just trying to think of ways this could blow up in weird ways.

Reply:
If it's anything else than a list/array/dataframe _with the same size as the times_, it fails at a later stage (it enters the calculation of the histogram and fails there).

Reply:
ok, cool, just making sure. :)